hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -8,14 +8,285 @@ title: "OTP Controller Technical Specification"
+
+Unfortunately, secret partitions must utilize a global netlist key for the scrambling operation, as there is no other non-volatile storage to store a unique key.
+
+
+## Partition Checks
+
+### Integrity
+
+Once the appropriate partitions have been locked, the hardware integrity checker employs two integrity checks to verifies the content of the volatile buffer registers:
+
+1. All buffered partitions have additional Byte parity protection that is concurrently monitored.
+2. The digest of the partition is recomputed at semi-random intervals and compared to the digest stored alongside the partition.
+
+The purpose of this check is NOT to check between the storage flops and the OTP, but whether the buffer register contents remain consistent with the calculated digest.
+This verification is primarily concerned with whether the storage flops have experienced fault attacks.
+This check applies to only the HW_CFG and SECRET partitions.
+If a failure is encountered, the OTP controller will send out an `otp_integrity_mismatch` alert and reset all of its hardware outputs to their defaults (**TBD do we need an LFSR clearing mechanism here for the secret partition?**)
+
+### Storage Consistency
+
+This verification ensures the value stored in the buffer registers remain consistent with those in the OTP.
+This process re-reads the OTP at semi-random intervals and confirms the value read is the same as the value stored.
+Note, given there are integrity checks in parallel, it is not necessary for some partitions to check ALL read contents for consistency.
+If there is an integrity digest, only the digest needs to be read; otherwise, all values must be read.
+
+
+This check applies to LIFE_CYCLE, HW_CFG and SECRET partitions.
+If a failure is encountered, the OTP controller will send out an `otp_consistency_mismatch` alert and reset all of its hardware outputs to their defaults (**TBD do we need an LFSR clearing mechanism here for the secret partition?**)
+
+Note that checks applied to life cycle could cause a failure if life cycle is updated, because life cycle is the only partition that may contain live updates.
+The controller hence detects this condition and makes sure that the buffer registers are kept up to date in order to prevent false positives.
+
+### Secret Partition Integrity Checks
+
+Since the secret partition is stored encrypted, this also implies the integrity digest is calculated over the encrypted form.
+In order to balance the amount of buffer registers needed, only the decrypted form of the secret partition is held in buffer registers.
+Hardware calculates the digest by re-encrypting the data before passing it through the digest.
+
+
+## Power-up and Sense
+
+The OTP controller partition storage must output a specified default (it is not always 0 like a blank OTP) upon reset release.
```

Comment:
@cdgori 
do you think it makes sense to elaborate more on the defaults here?
It should in theory be "safe" defaults, meaning things are "disabled" by default.

Reply:
o wait nvm, i guess the section below talks about this.

Reply:
I see what you mean... I've added the word "safe" before "default" in this sentence